app/src/main/java/org/gnucash/android/ui/settings/BackupPreferenceFragment.java

```diff
@@ -292,64 +296,75 @@ private void toggleOwnCloudSync(Preference pref){
 		}
 	}
 
-
 	public static GoogleApiClient getGoogleApiClient(final Context context) {
-		return new GoogleApiClient.Builder(context)
+		return getGoogleApiClient(context, null);
+	}
+
+	public static GoogleApiClient getGoogleApiClient(final Context context, OnCloudStorageListener listener) {
+		final WeakReference<OnCloudStorageListener> weakListener = new WeakReference<>(listener);
```

Comment:
the problem is that the function is "static". any references to heavy objects like fragments or activities will cause memory leaks.